changelog-generator/src/config.rs

```
@@ -0,0 +1,132 @@
+pub struct Config<'a> {
+    pub repo_path: Option<String>,
+    pub auth_pair: (&'a str, &'a str),
+    pub labels: LabelMap,
+    pub prefix_labels: LabelMap,
+    pub suffix_labels: LabelMap,
+    pub version_pattern: String,
+}
+
+impl<'a> Config<'a> {
+    pub fn new(args: &'a Vec<String>) -> Self {
+        let mut cli_repo_path = "";
+        let mut auth_pair = ("", "");
+        let mut config_path = "./config.json";
+
+        for idx in 1..args.len() {
+            match args[idx].as_str() {
+                "--config" | "-c" => {
+                    assert!(
+                        idx + 1 < args.len(),
+                        "Invalid Config Arguments ( --config/-c followed by the config path"
+                    );
+                    if Path::new(&args[idx + 1]).exists() {
+                        config_path = args[idx + 1].as_str();
+                    } else {
+                        println!(
+                            "{}",
+                            "Using default config location as the path given does not exist"
+                        );
+                    }
+                }
+                "--user" | "-u" => {
+                    assert!(
+                        idx + 2 < args.len(),
+                        "Invalid Config Arguments ( --config/-c followed by the config path"
+                    );
+                    auth_pair = (args[idx + 1].as_str(), args[idx + 2].as_str())
+                }
+                "--repo" | "-r" => {
+                    assert!(
+                        idx + 1 < args.len(),
+                        "Invalid Repo Arguments ( --repo/-r followed by the config path"
+                    );
+                    assert!(
+                        Path::new(&args[idx + 1]).exists(),
+                        "Repo Path does not exist"
+                    );
+                    cli_repo_path = args[idx + 1].as_str();
+                }
+                _ => {}
```

I think it's better to throw out some panic exception when some invalid param is passed.
It can also help find unexpected behaviors caused by spelling mistakes.

---

You are right, ill add it